pkg/volume/local_volume/local_volume.go

```diff
@@ -201,6 +202,34 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 
+	// If fsGroup is not nil, we need to make sure the requested fsGroup doesn't
+	// conflict with fsGroup already set for previous pod.
+	if fsGroup != nil {
```

Comment:
Double checked; if the 2nd pod has no fsGroup, then it runs as root, but SetVolumeOwnership won't do anything if fsGroup is nil, which means the the volume remains the same fsGroup from pod1 and both can write to that volume.